Lib/test/test__colorize.py

```diff
@@ -0,0 +1,43 @@
+import contextlib
+import sys
+import traceback
+import unittest
+import unittest.mock
+import _colorize
+from test.support import captured_output
+
+
+class TestColorizedTraceback(unittest.TestCase):
+    def test_colorized_detection_checks_for_environment_variables(self):
+        if sys.platform == "win32":
+            virtual_patching = unittest.mock.patch(
+                "nt._supports_virtual_terminal", return_value=True
+            )
+        else:
+            virtual_patching = contextlib.nullcontext()
+
+        env_vars_expected = [
+            ({'TERM': 'dumb'}, False),
+            ({'PYTHON_COLORS': '1'}, True),
+            ({'PYTHON_COLORS': '0'}, False),
+            ({'NO_COLOR': '1'}, False),
+            ({'NO_COLOR': '1', "PYTHON_COLORS": '1'}, True),
+            ({'FORCE_COLOR': '1'}, True),
+            ({'FORCE_COLOR': '1', 'NO_COLOR': '1'}, False),
+            ({'FORCE_COLOR': '1', "PYTHON_COLORS": '0'}, False),
+        ]
+
+        with virtual_patching:
+            with unittest.mock.patch("os.isatty") as isatty_mock:
+                isatty_mock.return_value = True
+
+                for env_vars, expected in env_vars_expected:
+                    with unittest.mock.patch("os.environ", env_vars):
+                        self.assertEqual(_colorize.can_colorize(), expected)
```

Comment:
If we use `subTest`, then it's nice and clear exactly which iteration of the `for` loop failed (if one does fail):

```pytb
======================================================================
FAIL: test_colorized_detection_checks_for_environment_variables (test.test__colorize.TestColorizedTraceback.test_colorized_detection_checks_for_environment_variables) (env_vars={'PYTHON_COLORS': '1'}, expected_color=False)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/alexw/dev/cpython/Lib/test/test__colorize.py", line 37, in test_colorized_detection_checks_for_environment_variables
    self.assertEqual(_colorize.can_colorize(), expected)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: True != False

----------------------------------------------------------------------
```

```suggestion
                    with self.subTest(env_vars=env_vars, expected_color=expected):
                        with unittest.mock.patch("os.environ", env_vars):
                            self.assertEqual(_colorize.can_colorize(), expected)
```

Reply:
Nice, I've not used `subTest` before, looks a bit like `pytest.mark.parametrize` 👍

Reply:
Subtests are great! In some ways, I actually prefer them to `parametrize` (though I much prefer pytest as a testing framework in general). There's a nice article on the differences between subtests and `parametrize` here: https://blog.ganssle.io/articles/2020/04/subtests-in-python.html